### PR TITLE
Switch random refinement to use only one FFT

### PR DESCRIPTION
**Pull request by smkuls**
* We can get away with doing only one polynomial evaluation for Random
Refinement.
* Updated the tests to test the same.

### CODE REVIEW

File: tests/progs/test_random_refinement.py [hunk 1]
```diff
@@ -1,18 +1,23 @@
...
+
+    async def _prog(context):
+        random_shares = [test_preprocessing.elements.get_rand(
+            context).v.value for i in range(k)]
+        refined_random_shares = refine_randoms(n, t, galois_field, random_shares)
+        assert len(refined_random_shares) == k-t
+        randoms = await context.ShareArray(refined_random_shares).open()
+        return tuple(randoms)
+
+    program_runner = TaskProgramRunner(n, t)
```
Comment (Drake-Eidukas) [inline tests/progs/test_random_refinement.py]: BTW, you can cut out this boilerplate with the `test_runner` fixture.

**Review by Drake-Eidukas — CHANGES_REQUESTED** [review]
Comment on boilerplate, otherwise this looks clean.

File: tests/progs/test_random_refinement.py (hunk 1 above)
Comment (smkuls) [reply tests/progs/test_random_refinement.py]: Done!

**Review by Drake-Eidukas — APPROVED** [review]
LGTM, just rebase and this is good to go.

**Comment by amiller** [comment]
yup its good

**Pull request merged**